### PR TITLE
Add a method to sample a dataset by ratio

### PR DESCRIPTION
**Pull request by shonohs**

### CODE REVIEW

File: tests/test_create_dataset_manifest.py [hunk 1]
```diff
@@ -346,6 +346,37 @@ def test_multitask(self):
         assert _get_instance_count_per_class(few_shot_manifest) == {2: 13, 3: 13, 0: 11, 4: 11, 1: 10, 5: 10}
 
 
+class TestManifestSubsetByRatio(unittest.TestCase):
```
Comment (pj-ms) [inline tests/test_create_dataset_manifest.py:349]: should we add a test for detection as well?

File: vision_datasets/common/data_manifest.py [hunk 2]
```diff
@@ -288,6 +289,41 @@ def sample_few_shot_subset(self, num_samples_per_class, random_seed=0):
...
+        if self.is_multitask:
+            labels = [[self._get_cid(c, t) for t, t_labels in image.labels.items() for c in t_labels] for image in self.images]
+        else:
+            labels = [[self._get_cid(c) for c in image.labels] for image in self.images]
+
+        # Create a dict {label_id: [image_id, ...], ...}
+        # Note that image_id can be included multiple times if the dataset is multilabel, objectdetection, or multitask.
+        label_image_map = collections.defaultdict(list)
+        for i, image_labels in enumerate(labels):
+            if not image_labels:
```
Comment (pj-ms) [inline vision_datasets/common/data_manifest.py:314]: there is a _is_negative() function which takes care of multitask case as well

File: tests/test_create_dataset_manifest.py (hunk 1 above)
Comment (shonohs) [reply tests/test_create_dataset_manifest.py:349]: good catch. added it.

File: vision_datasets/common/data_manifest.py (hunk 2 above)
Comment (shonohs) [reply vision_datasets/common/data_manifest.py:314]: this "labels" variable is a 2-d list constructed above. image_labels will be a 1-d empty list even on multitask case.

Comment (pj-ms) [reply vision_datasets/common/data_manifest.py:314]: got it

**Review by pj-ms — APPROVED** [review]

**Pull request merged**